### ISSUE TITLE
azurerm_resource_provider_registration support for Management Groups

### ISSUE THREAD

**Issue by danielfears** [hashicorp/terraform-provider-azurerm#22760]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment and review the [contribution guide](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/README.md) to help.

### Description

Some resources, such as Azure Virtual Network Manager, require a provider to be registered at the Management Group scope as outlined here: https://learn.microsoft.com/en-us/azure/virtual-network-manager/concept-network-manager-scope#scope

But it looks like this Terraform resource only covers registering providers at the Subscription scope.

I would like the option to provide an additional parameter to target provider registration to Management Groups, as currently the only way I can find to achieve this, is to manually run a POST request, as per the following docs: https://learn.microsoft.com/en-us/rest/api/resources/providers/register-at-management-group-scope

### New or Affected Resource(s)/Data Source(s)

azurerm_resource_provider_registration

### Potential Terraform Configuration

```hcl
resource "azurerm_resource_provider_registration" "example" {
  name = "Microsoft.Networks"
  scope {
    subscription_ids = [data.azurerm_subscription.current.id]
    management_group_ids = [data.azurerm_management_group.mg.id]
  }
}
```

**Comment by teowa** [comment]
Hi @danielfears , thanks for submitting this!
For now it is hard to integrate this into Terraform because there is only one method [Providers - Register At Management Group Scope](https://learn.microsoft.com/en-us/rest/api/resources/providers/register-at-management-group-scope), we cannot retrieve the registration status or unregister it.
workarounds:
- embed Az command into Terraform resource:
```hcl
data "azurerm_management_group" "example" {
  name = "00000000-0000-0000-0000-000000000000"
}

resource "null_resource" "register_rp_to_mg" {
  provisioner "local-exec" {
    command = <<CMD
         az provider register --namespace 'Microsoft.Network' -m ${data.azurerm_management_group.example.name}
    CMD
  }
}
```

**Comment by cschar** [comment]
looking forward for this addition

was trying to follow eventgrid tutorial here
https://learn.microsoft.com/en-us/azure/governance/policy/tutorials/route-state-change-events#create-an-event-grid-system-topic

and part of it was managing providers at mg level

**Comment by Netkracker** [comment]
should be possible to workaround the limitation using azapi see: https://github.com/Azure/terraform-provider-azapi/issues/299

**Comment by moler-udi** [comment]
@Netkracker, did you get it to work with the woraround?

For me I just get an error trying to register provider on mgmt group level.

Seems like there was a now closed issue on it:
https://github.com/Azure/azure-sdk-for-go/issues/21319

**Comment by Netkracker** [comment]
yes we solved the problem using azapi, we needed to loop over multiple management groups here a code sample:

resource "azapi_resource_action" "resource_providers" {
  for_each = { for mg in var.avnm_managed_management_groups : split("/", mg)[4] => mg }

  type        = "Microsoft.Management/managementGroups@2021-04-01"
  resource_id = "/providers/Microsoft.Management/managementGroups/${each.key}"
  action      = "providers/Microsoft.Network/register"
  method      = "POST"
}

**Comment by moler-udi** [comment]
@Netkracker, thanks got it working with your example.

**Comment by Jingwei-MS** [comment]
Hi community, the current Azure API doesn’t yet provide full support for managing registration at the Management Group level, so Terraform can’t fully integrate it at this point. Appreciate the workaround shared here, it’s super helpful for the community.
